File: src/main/java/com/cenfotec/disenio/conceptual/software/pov/app/cenfotecdisenioconceptualsoftwarepovapp/domain/Discount.java
package com.cenfotec.disenio.conceptual.software.pov.app.cenfotecdisenioconceptualsoftwarepovapp.domain;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;

import java.util.List;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection= "discounts")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Discount {

    @Id
    private String  id;
    private String description;
    private double percentage;
    private List<Document> documents;

}

File: frontend/patrones-facturacion/src/app/Models/Discount.ts
export class Discount {
    id: String = "0";
    description: string = "";
    percentage: number = 0;

    constructor(
        id?: string ,
        description?: string,
        percentage?: number
    ){
        this.id = id ||"0";
        this.description = description || "";
        this.percentage = percentage || 0;
    }



}
